node_modules/oae-admin/createtenant/js/createtenant.js

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright 2015 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['jquery', 'oae.core'], function($, oae) {
+
+    return function(uid, showSettings) {
+
+        // The widget container
+        var $rootel = $('#' + uid);
+
+        /**
+         * Set up the validation on the `create tenant` form, including the error messages
+         */
+        var setUpValidation = function() {
+            var validateOpts = {
+                'rules': {
+                    'host': {
+                        'hostname': true
+                    }
+                },
+                'messages': {
+                    'alias': {
+                        'required': oae.api.i18n.translate('__MSG__PLEASE_ENTER_A_TENANT_ALIAS__', 'createtenant')
+                    },
+                    'displayName': {
+                        'required': oae.api.i18n.translate('__MSG__PLEASE_ENTER_A_TENANT_NAME__')
+                    },
+                    'host': {
+                        'required': oae.api.i18n.translate('__MSG__PLEASE_ENTER_A_HOST_NAME__'),
+                        'hostname': oae.api.i18n.translate('__MSG__PLEASE_ENTER_A_VALID_HOST_NAME__')
+                    }
+                },
+                'methods': {
+                    'hostname': {
+                        'method': oae.api.util.validation().isValidHost
+                    }
+                },
+                'submitHandler': createTenant
+            };
+            oae.api.util.validation().validate($('#createtenant-form', $rootel), validateOpts);
+        };
+
+        /**
+         * Create a new tenant
+         */
+        var createTenant = function() {
+            var alias = $.trim($('#createtenant-alias', $rootel).val());
```

Comment:
I seem to remember that we have a form serialization utility that will extract the values from a form (and do trimming). Given that we'll be adding fields to this form, that might be useful to use here. However, I can't immediately find where that utility is ...